drivers/usb/udc/udc_dwc2_vendor_quirks.h

```diff
@@ -308,6 +308,105 @@ DT_INST_FOREACH_STATUS_OKAY(QUIRK_NRF_USBHS_DEFINE)
 
 #endif /*DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_usbhs_nrfs) */
 
+#if DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_usbhs_usbreg)
+
+#define DT_DRV_COMPAT snps_dwc2
+
+#define USBHS_DT_WRAPPER_REG_ADDR(n) UINT_TO_POINTER(DT_INST_REG_ADDR_BY_NAME(n, wrapper))
+
+/*
+ * On USBHS, we cannot access the DWC2 register until VBUS is detected and
+ * valid. If the user tries to force usbd_enable() and the corresponding
+ * udc_enable() without a "VBUS ready" notification, the event wait will block
+ * until a valid VBUS signal is detected or until the
+ * CONFIG_UDC_DWC2_USBHS_VBUS_READY_TIMEOUT timeout expires.
+ */
+static K_EVENT_DEFINE(usbhs_events);
+#define USBHS_VBUS_READY	BIT(0)
+
+static inline int usbhs_enable_usbreg(const struct device *dev)
+{
+	LOG_WRN("VBUS detection not implemented.");
+	k_event_post(&usbhs_events, USBHS_VBUS_READY);
+	udc_submit_event(dev, UDC_EVT_VBUS_READY, 0);
+
+	return 0;
+}
+
+static inline int usbhs_enable_core(const struct device *dev)
+{
+	NRF_USBHS_Type *wrapper = USBHS_DT_WRAPPER_REG_ADDR(0);
+
+	if (!k_event_wait(&usbhs_events, USBHS_VBUS_READY, false, K_NO_WAIT)) {
+		LOG_WRN("VBUS is not ready, block udc_enable()");
+		if (!k_event_wait(&usbhs_events, USBHS_VBUS_READY, false, K_FOREVER)) {
+			return -ETIMEDOUT;
+		}
+	}
+
+	wrapper->ENABLE = USBHS_ENABLE_PHY_Msk | USBHS_ENABLE_CORE_Msk;
+	wrapper->TASKS_START = 1UL;
+
+	/* Enable interrupts */
+	wrapper->INTENSET = 1UL;
+
+	return 0;
+}
+
+static inline int usbhs_disable_core(const struct device *dev)
+{
+	NRF_USBHS_Type *wrapper = USBHS_DT_WRAPPER_REG_ADDR(0);
+
+	/* Disable interrupts */
+	wrapper->INTENCLR = 1UL;
+
+	wrapper->ENABLE = 0UL;
+	wrapper->TASKS_START = 1UL;
+
+	return 0;
+}
+
+static inline int usbhs_irq_clear(const struct device *dev)
+{
+	NRF_USBHS_Type *wrapper = USBHS_DT_WRAPPER_REG_ADDR(0);
+
+	wrapper->EVENTS_CORE = 0UL;
+
+	return 0;
+}
+
+static inline int usbhs_init_caps(const struct device *dev)
+{
+	struct udc_data *data = dev->data;
+
+	data->caps.can_detect_vbus = true;
+	data->caps.hs = true;
+
+	return 0;
+}
```

Comment:
The existing can be move within `#if DT_HAS_COMPAT_STATUS_OKAY(x) && DT_HAS_COMPAT_STATUS_OKAY(y) ... #endif` and reused.

Reply:
I think you meant `||` instead of `&&`.